rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -1488,6 +1535,118 @@ RMW_Connext_Subscriber::take_serialized(
   return rc;
 }
 
+rmw_ret_t
+RMW_Connext_Subscriber::set_cft_expression_parameters(
+  const char * filter_expression,
+  const rcutils_string_array_t * expression_parameters)
+{
+  RMW_CONNEXT_ASSERT(nullptr != filter_expression)
+  RMW_CONNEXT_ASSERT(!internal)
+  RMW_CONNEXT_ASSERT(nullptr != node)
+
+  rmw_ret_t ret;
+  std::lock_guard<std::mutex> lock(this->cft_mutex);
+  const bool filter_expression_empty = (*filter_expression == '\0');
+  if (nullptr == this->dds_topic_cft && filter_expression_empty) {
+    // allow to call set filter_expresson even if cft not exist
+    RMW_CONNEXT_LOG_DEBUG("current subscriber has no content filter topic")
+    return RMW_RET_OK;
+  } else if (nullptr != this->dds_topic_cft && !filter_expression_empty) {
+    // set filter expre if cft exist
+    return rmw_connextdds_set_cft_filter_expression(
+      this->dds_topic_cft, filter_expression, expression_parameters);
+  }
+
+  // finalization to remove the old data reader
+  ret = finalize(true /* reset_cft */);
+  if (RMW_RET_OK != ret) {
+    RMW_CONNEXT_LOG_ERROR_SET("failed to finalize subscriber with resetting cft flag")
+    return ret;
+  }
+  DDS_DomainParticipant * const dp = this->dds_participant();
+  DDS_TopicDescription * sub_topic = nullptr;
+  if (nullptr == this->dds_topic_cft) {
+    // create a new cft topic, and then use cft topic to create a new reader
+    std::string cft_topic_name =
+      fqtopic_name + "_ContentFilterTopic" + RMW_Connext_Subscriber::get_atomic_id();
+
+    ret = rmw_connextdds_create_contentfilteredtopic(
+      ctx, dp, dds_topic, cft_topic_name.c_str(),
+      filter_expression,
+      expression_parameters, &sub_topic);
+
+    if (RMW_RET_OK != ret) {
+      RMW_CONNEXT_LOG_ERROR("failed to create content filter topic")
+      return ret;
+    }
+
+    this->dds_topic_cft = sub_topic;
+  } else {
+    // delete cft, and then use the existing parent topic to create a new reader
+    ret = rmw_connextdds_delete_contentfilteredtopic(
+      ctx, dp, this->dds_topic_cft);
+
+    if (RMW_RET_OK != ret) {
+      return ret;
+    }
+    this->dds_topic_cft = nullptr;
+
+    sub_topic = DDS_Topic_as_topicdescription(dds_topic);
+  }
+
+  DDS_DataReader * dds_reader = initialize_datareader(
```

Comment:
Please qualify the arguments to this function with `this->` to make it clear that we are accessing instance members/methods.

This is of course a "coding style" change to make the code consistent, more than a functional one. I'm not sure if qualifying every access with `this->` is the "modern C++ programmer's way", but I've been consistent so far in doing so (because I personally think it makes code much more readable). We should be consistent with that strategy (as flawed as it may be) until the moment we decide to not do it any more, and update all the code base accordingly.

I will try to point out some other occurrences of this, but please take a pass at the code to make it consistent with this rule.

Reply:
Also, qualify the function call as `RMW_Connext_Subscriber::initialize_datareader(`